src/enumerable.rb

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    length.times do |i|
+      yield(self[i])
+    end
+  end
+
+  def my_each_with_index
+    length.times do |i|
+      yield(self[i], i)
+    end
+  end
+
+  def my_select
+    selected_items = []
+    my_each do |item|
+      selected_items << item if yield(item)
+    end
+    selected_items
+  end
+
+  def my_all?
+    my_each { |item| return false unless yield(item) }
+    true
+  end
+
+  def my_any?
+    my_each { |item| return true if yield(item) }
+    false
+  end
+
+  def my_none?
+    my_each { |item| return false if yield(item) }
+    true
+  end
```

Comment:
#### 12. , 13. , 14. , 15. 
:recycle:  All the listed behaviors in points 4,5,6, 7,8,9,10,11  for the method my_all? and my_any? are failing here too.